dialogue-core/src/main/java/com/palantir/dialogue/core/RunningTimersLongImpl.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.github.benmanes.caffeine.cache.Ticker;
+
+final class RunningTimersLongImpl implements RunningTimers {
+
+    private final Ticker ticker;
+
+    // TODO(dfox): binpack both of these into an AtomicLong
```

Comment:
if we like the overall idea of this PR, then I'll put together a threadsafe implementation of this and actually write some tests to check it has sufficient capacity